app/share/views.py

```diff
@@ -88,6 +96,332 @@ def resources():
 @share.route('/files/<model_run_uuid>')
 @login_required
 def files(model_run_uuid):
+
+    if not os.path.exists(app.config['UPLOAD_FOLDER']):
+        os.makedirs(app.config['UPLOAD_FOLDER'])
+
     "View of file submission for as yet unselected resource to add to"
     model_run_uuid = model_run_uuid
     return render_template('share/files.html', model_run_uuid=model_run_uuid)
+
```

Comment:
The following upload function should be general. That is, any reference to PRMS should be removed and this should be simplified to only take one or a set of files when the user clicks the `Browse...` button.